tools/tsp-client/src/fs.ts

```diff
@@ -47,3 +48,48 @@ export async function tryReadTspLocation(rootDir: string): Promise<string | unde
   }
   return undefined;
 }
+
+export async function readTspLocation(rootDir: string): Promise<[string, string, string, string[]]> {
+  try {
+    const yamlPath = path.resolve(rootDir, "tsp-location.yaml");
+    const fileStat = await stat(yamlPath);
+    if (fileStat.isFile()) {
+      const fileContents = await readFile(yamlPath, "utf8");
+      const locationYaml = parseYaml(fileContents);
+      var { directory, commit, repo, additionalDirectories } = locationYaml;
+      if (!directory || !commit || !repo) {
+        throw new Error("Invalid tsp-location.yaml");
+      }
+      Logger.info(`Additional directories: ${additionalDirectories}`)
+      if (!additionalDirectories) {
+        additionalDirectories = [];
+      }
+      return [ directory, commit, repo, additionalDirectories ];
+    }
+  } catch (e) {
+    Logger.error(`Error reading tsp-location.yaml: ${e}`);
+  }
+  throw new Error("Could not find tsp-location.yaml");
+}
+
+
+export async function findEmitterPackage(emitterPath: string): Promise<string | undefined> {
+  if (existsSync(emitterPath)) {
+    const emitter = await readFile(emitterPath, 'utf8').then(data => {
```

Comment:
please don't mix `await` and promise chaining with `then`, it can create some headaches.